modules/adfs/lib/IdP/ADFS.php

```diff
@@ -157,7 +164,7 @@ public static function sendResponse(array $state) {
 		$wresult = sspmod_adfs_IdP_ADFS::ADFS_SignResponse($response, $privateKeyFile, $certificateFile);
 
 		$wctx = $state['adfs:wctx'];
-		sspmod_adfs_IdP_ADFS::ADFS_PostResponse($spMetadata->getValue('prp'), $wresult, $wctx);
+		sspmod_adfs_IdP_ADFS::ADFS_PostResponse(($state['adfs:wreply']) ? $state['adfs:wreply'] : $spMetadata->getValue('prp'), $wresult, $wctx);
```

Comment:
The key is set to an empty string when unavailable, which renders to a false value.. Explicit make is 'false' is perhaps better..